astronomy/лидов_古在_test.cpp

```diff
@@ -0,0 +1,165 @@
+                /*step=*/10 * Minute))),
+        mercury_(*solar_system_1950_.massive_body(*ephemeris_, "Mercury")),
+        mercury_frame_(ephemeris_.get(), &mercury_) {}
+
+  SolarSystem<ICRS> solar_system_1950_;
+  not_null<std::unique_ptr<Ephemeris<ICRS>>> ephemeris_;
+  MassiveBody const& mercury_;
+  BodyCentredNonRotatingDynamicFrame<ICRS, MercuryCentredInertial>
+      mercury_frame_;
+};
+
+#if !_DEBUG
+TEST_F(Лидов古在Test, MercuryOrbiter1) {
+  DiscreteTrajectory<ICRS> icrs_trajectory;
+  icrs_trajectory.Append(MercuryOrbiter1119InitialTime,
+                         MercuryOrbiter1119InitialDegreesOfFreedom<ICRS>);
+  icrs_trajectory.SetDownsampling(ksp_plugin::MaxDenseIntervals,
+                                  ksp_plugin::DownsamplingTolerance);
+  auto const instance = ephemeris_->NewInstance(
+      {&icrs_trajectory},
+      Ephemeris<ICRS>::NoIntrinsicAccelerations,
+      {SymmetricLinearMultistepIntegrator<Quinlan1999Order8A, Position<ICRS>>(),
+       /*step=*/10 * Second});
+
+  for (int year = 1967; year < 1980; ++year) {
+    Instant const t = DateTimeAsTT(date_time::DateTime::BeginningOfDay(
+        date_time::Date::Ordinal(year, 1)));
+    LOG(ERROR) << "Flowing to " << t;
+    auto const status = ephemeris_->FlowWithFixedStep(t, *instance);
+    if (!status.ok()) {
+      LOG(ERROR) << status << " at " << icrs_trajectory.back().time;
+      break;
+    }
+  }
+
+  DiscreteTrajectory<MercuryCentredInertial> mercury_centred_trajectory;
+  for (auto const& [t, dof] : icrs_trajectory) {
+    mercury_centred_trajectory.Append(t,
+                                      mercury_frame_.ToThisFrameAtTime(t)(dof));
+  }
+
+  OrbitalElements const elements = OrbitalElements::ForTrajectory(
+      mercury_centred_trajectory, mercury_, MasslessBody{}).value();
+  mathematica::Logger logger(SOLUTION_DIR / "mathematica" / "1119.generated.wl",
+                             /*make_unique=*/false);
+  // The constants c₁ and c₂ are defined in [Лид61], equations (58) and (59)
+  // respectively.
+  Interval<double> c₁;
+  Interval<double> c₂;
+  for (auto const elements : elements.mean_elements()) {
+    double const ε = 1 - Pow<2>(elements.eccentricity);
+    double const cos²_i = Pow<2>(Cos(elements.inclination));
+    double const sin²_i = Pow<2>(Sin(elements.inclination));
+    double const sin²_ω = Pow<2>(Sin(elements.argument_of_periapsis));
+    c₁.Include(ε * cos²_i);
+    c₂.Include((1 - ε) * (2.0 / 5.0 - sin²_i * sin²_ω));
+    logger.Append("t", elements.time, mathematica::ExpressIn(Second));
+    logger.Append("a", elements.semimajor_axis, mathematica::ExpressIn(Metre));
+    logger.Append("e", elements.eccentricity);
+    logger.Append("i", elements.inclination, mathematica::ExpressIn(Radian));
+    logger.Append(R"(\[Omega])",
+                  elements.argument_of_periapsis,
+                  mathematica::ExpressIn(Radian));
+  }
+  // The elements e, i, and ω all vary quite a lot.
```

Comment:
The numbers would be easier to interpret if, instead of `min` and `max` (or in addition), you looked at the relative error between `measure` and `midpoint`.  It's like 75% for `e` and 17% for `c₁`.

Reply:
I don’t think the relative errors are meaningful here, since the point (c₁, c₂) lies within a bounded region within [0, 1] × [-3/5, 2/5] (depicted in [Лид61, p. 29], [LC62, p. 745]).

Reply:
How do you demonstrate that something is constant if you have lots of measurements with noise?  By looking at its statistical elements.  If you are not happy with a cheesy relative error, please do a proper statistical analysis.

Reply:
As discussed, what we care about is the absolute scale of the variations; they are not really noise, but they are very small compared to the size of the region given in Рис. 1.